auth-service/src/main/kotlin/uk/gov/communities/delta/auth/controllers/external/DeltaUserRegistrationController.kt

```diff
@@ -0,0 +1,207 @@
+package uk.gov.communities.delta.auth.controllers.external
+
+import OrganisationService
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import io.ktor.server.thymeleaf.*
+import org.slf4j.LoggerFactory
+import uk.gov.communities.delta.auth.config.AuthServiceConfig
+import uk.gov.communities.delta.auth.config.AzureADSSOConfig
+import uk.gov.communities.delta.auth.config.DeltaConfig
+import uk.gov.communities.delta.auth.oauthClientLoginRouteWithEmail
+import uk.gov.communities.delta.auth.services.Registration
+import uk.gov.communities.delta.auth.services.RegistrationService
+import uk.gov.communities.delta.auth.utils.EmailAddressChecker
+
+class DeltaUserRegistrationController(
+    private val deltaConfig: DeltaConfig,
+    private val authServiceConfig: AuthServiceConfig,
+    private val ssoConfig: AzureADSSOConfig,
+    organisationService: OrganisationService,
+    private val registrationService: RegistrationService,
+) {
+    private val logger = LoggerFactory.getLogger(this.javaClass)
+    private val emailAddressChecker = EmailAddressChecker(organisationService)
+
+    fun registerFormRoutes(route: Route) {
+        route.post {
+            registerPost(call)
+        }
+        route.get {
+            registerGet(call)
+        }
+    }
+
+    fun registerSuccessRoute(route: Route) {
+        route.get {
+            call.respond(
+                ThymeleafContent(
+                    "registration-success",
+                    mapOf(
+                        "deltaUrl" to deltaConfig.deltaWebsiteUrl,
+                        "emailAddress" to call.parameters["emailAddress"]!!,
+                    )
+                )
+            )
+        }
+    }
+
+    private suspend fun registerGet(call: ApplicationCall) {
+        call.respondRegisterPage()
+    }
+
+    private val firstNameEmpty = "First name must not be empty"
+    private val lastNameEmpty = "Last name must not be empty"
+    private val emailAddressEmpty = "Email address must not be empty"
+    private val confirmEmailAddressEmpty = "Confirm email address must not be empty"
+    private val notEqualEmails = "Email addresses do not match"
+    private val notAnEmailAddress = "Email address must be a valid email address"
+    private val notAKnownDomain =
+        "Email address domain not recognised, please use an email address associated with an organisation using DELTA"
```

Comment:
Should say something about contacting the service desk if they think it should work